`Straight_skeleton_2` can't be copy constructed due to private method in base halfedge class

## Issue Details

When attempting to copy an instance of [`Straight_skeleton_2`](https://doc.cgal.org/latest/Straight_skeleton_2/classCGAL_1_1Straight__skeleton__2.html), I'm met with an error regarding the base halfedge class' `set_next` method being private. When the `Straight_skeleton_2` base class' copy constructor is called (`HalfedgeDS_list`), it calls `pointer_update`, private within the class. So far, so good. Within this method, a polymorphic call to `set_next` is made, which, in the context of a `Straight_skeleton_2` is private. I'm guessing this is so to prevent mutation, as per the documentation (which entails this class is basically just an immutable wrapper around the default `HalfedgeDS`).

All in all, I'm not sure this is intended behavior, but at the very least, it goes against the documentation for `Straight_skeleton_2` where it is said it is a model of `CopyConstructible`.

To provide some context: this issue surfaced while attempting to map the 2D Straight Skeleton package to another target language - julia - [using a library](https://github.com/JuliaInterop/libcxxwrap-julia) that, when adding a type, creates a `copy` method given the type is `CopyConstructible`, which is asserted during compilation resorting to `std::is_copy_constructible`. Alas, that is not trivially the case and my code won't compile as a result.

## Source Code

Here is an MRE using CGAL on its own:

```cpp
#include <CGAL/Exact_predicates_inexact_constructions_kernel.h>
#include <CGAL/Straight_skeleton_2.h>

using Sk = CGAL::Straight_skeleton_2<CGAL::Epick>;

int main() {
  Sk s1;
  Sk s2(s1);
  return 0;
}

```

Compilation error log:

```
In file included from /usr/local/include/CGAL/HalfedgeDS_default.h:21,
                from /usr/local/include/CGAL/Straight_skeleton_2.h:20,
                from test.cpp:2:
/usr/local/include/CGAL/HalfedgeDS_list.h: In instantiation of ‘void CGAL::HalfedgeDS_list<Traits_, HalfedgeDSItems, Alloc>::pointer_update(const Self&) [with Traits_ = CGAL::Epick; HalfedgeDSItems = CGAL::Straight_skeleton_items_2; Alloc = std::allocator<int>; CGAL::HalfedgeDS_list<Traits_, HalfedgeDSItems, Alloc>::Self = CGAL::HalfedgeDS_list<CGAL::Epick, CGAL::Straight_skeleton_items_2, std::allocator<int> >]’:
/usr/local/include/CGAL/HalfedgeDS_list.h:403:9:   required from ‘CGAL::HalfedgeDS_list<Traits_, HalfedgeDSItems, Alloc>::HalfedgeDS_list(const Self&) [with Traits_ = CGAL::Epick; HalfedgeDSItems = CGAL::Straight_skeleton_items_2; Alloc = std::allocator<int>; CGAL::HalfedgeDS_list<Traits_, HalfedgeDSItems, Alloc>::Self = CGAL::HalfedgeDS_list<CGAL::Epick, CGAL::Straight_skeleton_items_2, std::allocator<int> >]’
/usr/local/include/CGAL/HalfedgeDS_default.h:30:7:   required from here
/usr/local/include/CGAL/HalfedgeDS_list.h:352:31: error: ‘void CGAL::Straight_skeleton_halfedge_base_2<Refs, S>::set_next(CGAL::Straight_skeleton_halfedge_base_2<Refs, S>::Halfedge_handle) [with Refs = CGAL::HalfedgeDS_list_types<CGAL::Epick, CGAL::Straight_skeleton_items_2, std::allocator<int> >; S = CGAL::Segment_2<CGAL::Epick>; CGAL::Straight_skeleton_halfedge_base_2<Refs, S>::Halfedge_handle = CGAL::internal::In_place_list_iterator<CGAL::HalfedgeDS_in_place_list_halfedge<CGAL::Straight_skeleton_halfedge_base_2<CGAL::HalfedgeDS_list_types<CGAL::Epick, CGAL::Straight_skeleton_items_2, std::allocator<int> >, CGAL::Segment_2<CGAL::Epick> > >, std::allocator<CGAL::HalfedgeDS_in_place_list_halfedge<CGAL::Straight_skeleton_halfedge_base_2<CGAL::HalfedgeDS_list_types<CGAL::Epick, CGAL::Straight_skeleton_items_2, std::allocator<int> >, CGAL::Segment_2<CGAL::Epick> > > > >]’ is private within this context
  352 |             h->HBase::set_next( h_map[ h->next()]);
      |             ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
In file included from /usr/local/include/CGAL/Straight_skeleton_vertex_base_2.h:18,
                 from /usr/local/include/CGAL/Straight_skeleton_items_2.h:17,
                 from /usr/local/include/CGAL/Straight_skeleton_2.h:19,
                 from test.cpp:2:
/usr/local/include/CGAL/Straight_skeleton_halfedge_base_2.h:131:8: note: declared private here
  131 |   void set_next    ( Halfedge_handle h )  { Base_base::set_next(h)  ; }
      |        ^~~~~~~~
```

## Environment

* Operating system (Windows/Mac/Linux, 32/64 bits): Linux 5.7.2-arch1-1 x86_64
* Compiler: g++ 10.1.0
* Release or debug mode: Both
* Specific flags used (if any): N/A
* CGAL version: 5.0.2
* Boost version: 1.72.0
* Other libraries versions if used (Eigen, TBB, etc.): N/A

--- a/Straight_skeleton_2/include/CGAL/Straight_skeleton_halfedge_base_2.h
+++ b/Straight_skeleton_2/include/CGAL/Straight_skeleton_halfedge_base_2.h
@@ -134,8 +134,6 @@ public:
 
   Straight_skeleton_halfedge_base_2( int aID, Sign aSlope ) : Base_base(aID,aSlope) {}
 
-private:
-
   void set_opposite( Halfedge_handle h )  { Base_base::opposite(h)  ; }
   void set_next    ( Halfedge_handle h )  { Base_base::set_next(h)  ; }
   void set_prev    ( Halfedge_handle h )  { Base_base::set_prev(h)  ; }

--- a/Straight_skeleton_2/test/Straight_skeleton_2/test_straight_skeleton_copy.cpp
+++ b/Straight_skeleton_2/test/Straight_skeleton_2/test_straight_skeleton_copy.cpp
@@ -1,0 +1,50 @@
+#include<vector>
+
+#include<boost/shared_ptr.hpp>
+
+#include<CGAL/Exact_predicates_inexact_constructions_kernel.h>
+#include<CGAL/Polygon_2.h>
+#include<CGAL/create_offset_polygons_2.h>
+
+typedef CGAL::Exact_predicates_inexact_constructions_kernel K ;
+
+typedef K::Point_2                   Point ;
+typedef CGAL::Polygon_2<K>           Polygon_2 ;
+typedef CGAL::Straight_skeleton_2<K> Ss ;
+
+typedef boost::shared_ptr<Polygon_2> PolygonPtr ;
+typedef boost::shared_ptr<Ss> SsPtr ;
+
+typedef std::vector<PolygonPtr> PolygonPtrVector ;
+
+int main()
+{
+  std::ifstream in("data/A.poly");
+  if (!in) return 1;
+  int n;
+  double x,y;
+  Polygon_2 poly ;
+
+  in >> n; // skip #polylines
+  in >> n;
+  for (int i=0; i<n; ++i)
+  {
+    in >> x >> y;
+    poly.push_back( Point(x,y) ) ;
+  }
+  // we are only taking the outer contour (sufficient for this test)
+
+  std::cout << poly.size() << "\n";
+
+  SsPtr ss = CGAL::create_interior_straight_skeleton_2(poly);
+  Ss ss_copy(*ss);
+
+  double lOffset = 5 ;
+
+  PolygonPtrVector offset_polygons = CGAL::create_offset_polygons_2<Polygon_2>(lOffset,ss_copy);
+
+  assert(offset_polygons.size()==1);
+  assert(offset_polygons.front()->size()>10);
+
+  return 0;
+}

